modules/pricer/src/main/java/com/opengamma/strata/pricer/calibration/MarketDataFxRateProvider.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.calibration;
+
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.currency.FxRate;
+import com.opengamma.strata.basics.currency.FxRateProvider;
+import com.opengamma.strata.basics.market.FxRateKey;
+import com.opengamma.strata.basics.market.MarketData;
+
+/**
+ * A provider of FX rates that takes its data from a {@link MarketData} instance.
+ */
+class MarketDataFxRateProvider implements FxRateProvider {
```

Comment:
I'm not sure. The `MarketData` API is deliberately close to `CalculationEnvironment` and `CalculationMarketData` and is basically a map of key to value. I'm not sure it's worth adding it just to support this one case. If we find ourselves using this class all the time we can reconsider.